src/main/java/app/persistence/MaterialMapper.java

```diff
@@ -13,11 +15,58 @@ public List<Material> getAllMaterials(String description, ConnectionPool connect
         return null;
     }
 
-    public void createMaterial(Material material, ConnectionPool connectionPool) {
+    public void createMaterial(Material material, ConnectionPool connectionPool) throws DatabaseException {
+        String sql = "INSERT INTO materials (material_id, material_description, material_unit, material_price) VALUES (?, ?, ?, ?)";
 
+        try (
+                Connection connection = connectionPool.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)
+        ) {
+            ps.setInt(1, material.getMaterialId());
```

Comment:
This is the same as the comment above. The id should not be provided manually as it is not known at the time and something the database will generate on its own